src/_pytest/_code/code.py

```diff
@@ -719,7 +719,9 @@ def repr_excinfo(self, excinfo):
             repr_chain = []
             e = excinfo.value
             descr = None
-            while e is not None:
+            seen = set()
+            while e is not None and id(e) not in seen:
```

Comment:
i believe for exceptions its fine to use the direct object banking on the object identity hashing by default,
and that way this wouldn't incur the same cost as it does with `id()` on pypy

Reply:
I didn't really want to make assumptions about exceptions since any user space exception could appear here, even one with an exceptionally bad `__hash__` + `__eq__`.  I considered using `object.__hash__(e)` instead of `id(e)` but I figure they're probably the same.

The "performance hit" in pypy is almost certainly negligible and this isn't a performance critical path (formatting failed test tracebacks).

Reply:
thanks for the rebuttal